### DISCUSSION TITLE
CotEditor & rmate (textmate server)

### DISCUSSION THREAD

**Discussion by gsmn2gsmn** [coteditor/CotEditor#1290] [Q&A]
Hi there,

I really like the easy syntax customization avaiable with your program 
and because of that I'd like to use it as my daily activities tool.

But I really need a remote editing feature like the one available with textmate (rmate).

I discovered a standalone server that duplicates the rmate server functionality:
https://github.com/xiaozhuai/rmate-server
so in theory it would be possibile to have the rmate functionality in CotEditor too,
with no effort.

With this server I'm able to edit remote files with atom  (I made a test to see if the server works) for example 
but not with CotEditor.

I've installed the cot python script via symlink and it works if used from the terminal,
permission in system preferences have been set to.

<img width="157" alt="immagine" src="https://user-images.githubusercontent.com/104671173/166096571-17f7669c-bc6c-4cee-8872-f7ddbee3f31f.png">

In my test, the file is moved from the server to my mac but then the CotEditor window
is not opened.

![layout-SQLPYTHON__Python__e_New_Discussion_·_coteditor_CotEditor_e_Sicurezza_e_Privacy](https://user-images.githubusercontent.com/104671173/166096654-0ed177a3-a497-4339-9472-3f9f85994175.jpg)

che needed config.json file si correct:

{
    "host": "127.0.0.1",
    "port": 52698,
    "editor": {
        "command": "cot",
        "args": [
            "-w",
            "{{file}}"
        ]
    }
}

other mac security check may block the cot program when called by the rmate server?
any ideas?

**Comment by radekkozak** [comment]
Hi, long-time CotEditor user here.

Maybe this will interest you 👉🏻 [cotmate - remote editing for CotEditor](https://github.com/radekkozak/cotmate)

I've been using it as my own solution to remote editing with CotEditor for quite some time now and it seems to work quite well, so i decided to polish it a little and put it out there. 

It's **rmate-compatible** - in other words it speaks the same protocol - so in essence after installing `cotmate` you should be up and running with your existing `rmate` set up on your remote servers.

Personally i've been using [pure-Bash version of rmate](https://github.com/aurora/rmate) for like forever, but i also tested it all with [official Ruby version of rmate](https://github.com/textmate/rmate) and it seems to be working A-OK as well.

Any feedback or problems you find would be greatly appreciated.

In any case, hope it will work for you @gsmn2gsmn

Cheers
